library/init/meta/expr.lean

```diff
@@ -212,23 +212,23 @@ is_constant_of (get_app_fn e) n
 meta def is_napp_of (e : expr) (c : name) (n : nat) : bool :=
 is_app_of e c ∧ get_app_num_args e = n
 
-meta def is_false (e : expr) : bool :=
-is_constant_of e `false
+meta def is_false : expr → bool
+| ``(false) := tt
+| _         := ff
 
+set_option eqn_compiler.max_steps 256
```

Comment:
The eqn compiler definitely needs a few more steps for these, but the output size looks still reasonable.